src/repositories/Settings/UnitTypes/UnitTypeRepository.php

```diff
@@ -89,38 +89,40 @@ public function countUnitTypeWithLabel(string $name, string $tenantId, string|in
           if (isset($excludeId))
             $query->where('id', '!=', $excludeId);
         })
-        ->where(function ($query) use ($tenantId) {
-          $query->where(UnitTypeModel::getTenantColumnName(), '=', $tenantId)
-            ->orWhereNull(UnitTypeModel::getTenantColumnName());
-        })
+        ->where(UnitTypeModel::getTenantColumnName(), '=', $tenantId)
     )();
   }
 
-  public function findMany(string $tenantId)
+  public function findMany(string $tenantId, int $page = 0): Promise
   {
     return async(
       fn () => $this->getQueryBuilder()
         ->whereNull('deleted_at')
-        ->where(function ($query) use ($tenantId) {
-          $query->where(UnitTypeModel::getTenantColumnName(), '=', $tenantId)
-            ->orWhereNull(UnitTypeModel::getTenantColumnName());
-        })
+        ->where(UnitTypeModel::getTenantColumnName(), '=', $tenantId)
+        ->limit(self::DEFAULT_PAGE_SIZE)
+        ->offset($page * self::DEFAULT_PAGE_SIZE)
         ->get()
         ->map(function ($row) {
           return UnitTypeMapper::modelToEntity(UnitTypeModel::fromStdclass($row));
         })
     )();
   }
 
-  public function create(UnitTypeMutationData $data, string $tenantId): int|string
+  public function create(MutationDataInterface $data, string $tenantId): string
   {
-    $newId = $this->getQueryBuilder()->insertGetId(
-      UnitTypeMapper::serializeCreate($data, $tenantId)
-    );
-    return $newId;
+    try {
+      $uuid = Uuid::uuid4()->toString();
```

Comment:
Creating UUID upfront to give it back to the user for UX and avoid any concurrency or needs of a transaction if we wanted to get it back afterwards. It simplifies everything, is safer and is faster.